### PR TITLE
Delete datasets

### PR DESCRIPTION
**Pull request by sakeerthy**

### CODE REVIEW

File: c/datasetjson.c [hunk 1]
```diff
@@ -1122,6 +1123,238 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
...
+  strcpy(tmpName, absolutePath);
+  char *token = strtok(tmpName, "'");
+  char justName[40] = "";
+  while (token != NULL)
+  {
+    strcpy(justName, token);
+    token = strtok(NULL, "'");
+  }
+  int rc = deleteCluster(justName);
+  printf("%d\n", rc);
```
Comment (sakeerthy) [inline c/datasetjson.c]: remove print

File: c/datasetjson.c [hunk 2]
```diff
@@ -1122,6 +1123,238 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
...
+  int fieldCount = defaultCSIFieldCount;
+  char **csiFields = defaultCSIFields;
+
+  csi_parmblock *returnParms = (csi_parmblock*)safeMalloc(sizeof(csi_parmblock),"CSI ParmBlock");
+
+  char tmpName[40] = "";
+  strcpy(tmpName, absolutePath);
+  char *token = strtok(tmpName, "'");
+  char justName[40] = "";
+  while (token != NULL)
```
Comment (sakeerthy) [inline c/datasetjson.c]: function that does this already exists

File: c/datasetjson.c [hunk 3]
```diff
@@ -1122,6 +1123,238 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
...
+  {
+    strcpy(justName, token);
+    token = strtok(NULL, "'");
+  }
+  
+  EntryDataSet *entrySet = returnEntries(justName, typesArg, datasetTypeCount, 
+                                         workAreaSizeArg, csiFields, fieldCount, 
+                                         NULL, NULL, returnParms); 
+  EntryData *entry = entrySet->entries[0];
```
Comment (sakeerthy) [inline c/datasetjson.c]: add check that makes sure only one is returned

File: c/datasetjson.c [hunk 4]
```diff
@@ -1122,6 +1123,238 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
...
+  }
+  jsonEnd(p);
+ 
+  finishResponse(response);
+  
+#endif /* __ZOWE_OS_ZOS */
+}
+
+
+bool isVsam(char* absolutePath) {
```
Comment (sakeerthy) [inline c/datasetjson.c]: change return type to char of the csi type

File: c/datasetjson.c [hunk 5]
```diff
@@ -1122,6 +1123,238 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
...
+                                         workAreaSizeArg, csiFields, fieldCount, 
+                                         NULL, NULL, returnParms); 
+  EntryData *entry = entrySet->entries[0];
+  
+  bool boolVsam = false;
+  if(entry) {
+    for (int i = 0; i < sizeof(vsamCSITypes); i++) {
+        if (entry->type == vsamCSITypes[i]) {
+            boolVsam = true;
+        }
```
Comment (sakeerthy) [inline c/datasetjson.c]: index of exists in utils.c

File: c/datasetjson.c [hunk 6]
```diff
@@ -1122,6 +1123,238 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
...
+
+    finishResponse(response);  
+  } else {
+    jsonPrinter *p = respondWithJsonPrinter(response);
+    setResponseStatus(response, 400, "ERROR");
+    setDefaultJSONRESTHeaders(response);
+    writeHeader(response);
+    jsonStart(p);
+    jsonAddString(p, "Response", 
+                  "Error occurred with the id cams trying to delete the dataset. " +
```
Comment (sakeerthy) [inline c/datasetjson.c]: IDCAMS not id cams

File: c/datasetjson.c [hunk 7]
```diff
@@ -1122,6 +1123,238 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
...
+    setDefaultJSONRESTHeaders(response);
+    writeHeader(response);
+    jsonStart(p);
+    jsonAddString(p, "Response", "VSAM dataset deleted");
+    jsonEnd(p);
+
+    finishResponse(response);  
+  } else {
+    jsonPrinter *p = respondWithJsonPrinter(response);
+    setResponseStatus(response, 400, "ERROR");
```
Comment (sakeerthy) [inline c/datasetjson.c]: 500

Comment (sakeerthy) [reply c/datasetjson.c]: match response to 500

Comment (sakeerthy) [reply c/datasetjson.c]: defined in http.h

File: c/datasetjson.c [hunk 8]
```diff
@@ -1122,6 +1123,238 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
...
+  char tmpName[40] = "";
+  strcpy(tmpName, absolutePath);
+  char *token = strtok(tmpName, "'");
+  char justName[40] = "";
+  while (token != NULL)
+  {
+    strcpy(justName, token);
+    token = strtok(NULL, "'");
+  }
+  int rc = deleteCluster(justName);
```
Comment (sakeerthy) [inline c/datasetjson.c]: test with aix to see if it deletes cluster of the aix or just the aix

File: c/datasetjson.c [hunk 9]
```diff
@@ -1122,6 +1123,238 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
...
+    
+    /* Free member name and dcb as they are no longer needed */
+    free24(belowMemberName, DATASET_MEMBER_NAME_LEN);
+    closeSAM(dcb, 0);
+    
+    if (stowReturnCode != 0) {
+      zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_SEVERE,
+              "error: stowReturnCode=%d, stowReasonCode=%d\n",
+              stowReturnCode, stowReasonCode);
+      respondWithError(response, HTTP_STATUS_INTERNAL_SERVER_ERROR, "Member could not be deleted");
```
Comment (sakeerthy) [inline c/datasetjson.c]: include member name

File: c/datasetjson.c [hunk 10]
```diff
@@ -1122,6 +1123,238 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
...
+
+  jsonPrinter *p = respondWithJsonPrinter(response);
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+ 
+  writeHeader(response);
+
+  jsonStart(p);
+  if (isMemberEmpty) {
+    jsonAddString(p, "msg", "Data set was deleted successfully");
```
Comment (sakeerthy) [inline c/datasetjson.c]: actual name should be included

File: c/datasetjson.c [hunk 11]
```diff
@@ -1122,6 +1123,238 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
...
+    memcpy(belowMemberName, memberName.value, DATASET_MEMBER_NAME_LEN);
+    
+    int stowReturnCode = 0, stowReasonCode = 0;  
+    stowReturnCode = bpamDeleteMember(dcb, belowMemberName, &stowReasonCode);
+    
+    /* Free member name and dcb as they are no longer needed */
+    free24(belowMemberName, DATASET_MEMBER_NAME_LEN);
+    closeSAM(dcb, 0);
+    
+    if (stowReturnCode != 0) {
```
Comment (jordanfilteau1995) [inline c/datasetjson.c]: Do "dynallocUnallocDatasetByDDName()" first.

File: c/datasetjson.c [hunk 12]
```diff
@@ -1122,6 +1123,238 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
...
+  setDefaultJSONRESTHeaders(response);
+ 
+  writeHeader(response);
+
+  jsonStart(p);
+  if (isMemberEmpty) {
+    jsonAddString(p, "msg", "Data set was deleted successfully");
+  }
+  else {
+    jsonAddString(p, "msg", "Data set member was deleted successfully");
```
Comment (sakeerthy) [inline c/datasetjson.c]: actual name

**Comment by 1000TurquoisePogs** [comment]
Latest change looks good.

**Pull request closed without merging**